cosign/README.md

```diff
@@ -0,0 +1,137 @@
+    - Verifiers: Individuals or systems responsible for checking that the digital signature attached to an artifact is valid and that the artifact remains unaltered.
+
+3. Trust Infrastructure and Public-Key Management
+    - Certificate Authority (CA): An entity within a Public-key Infrastructure (PKI) that is responsible for verifying identities and issuing digital certificates which associate a public key with the verified identity of a signer.
+    - Public Key: The publicly disclosable component of a pair of cryptographic keys used for asymmetric cryptography.
+    - Public-key Certificate: A public-key certificate binds a subject name to a public key value, along with information needed to perform certain cryptographic functions using that key.
+    - Public-key Infrastructure (PKI): A system of CAs (and, optionally, RAs and other supporting servers and agents) that perform some set of certificate management, archive management, key management, and token management functions for a community of users in an application of asymmetric cryptography. The core PKI functions are (a) to register users and issue their public-key certificates, (b) to revoke certificates when required, and (c) to archive data needed to validate certificates at a much later time.
+
+4. Transparency and Auditability
+    - Identity Log: An online provided record, that maps verified identities to their associated signing keys. Such logs help establish trust by providing a history of key associations.
+    - Artifact Log: An online/offline provided record that contains metadata about artifacts, such as when and by whom they were signed. This ledger aids in auditing and tracing the provenance of digital artifacts.
+    - Timestamps: Time is a critical component of Sigstore. It’s used to verify that a short-lived certificate issued by Fulcio was valid at a previous point, when the artifact was signed.
+
+## Scenarios
+
+`ratify-verifier-go` supports multiple verification scenarios based on different signing methods, artifact types, and trust sources. These scenarios can be categorized into the following main types.
+
+| Verification Scenario             | Purpose                                                                | Use Case                                                                    |
+|-----------------------------------|------------------------------------------------------------------------|-----------------------------------------------------------------------------|
+| **Keyless Verification**          | Verifies signatures from Rekor, trust material (signed certificate timestamp) from the CT log, and certificates that chain up to Fulcio. | CI/CD pipelines or automated workflows where no private key management is needed. |
+| **Key-Based Verification**        | Verifies signature using a known **public key**.                        | Using an on-disk public key, or a key provided by other providers (e.g., "KMS provider") |
```

Comment:
and the scenario should be considered from the Ratify perspective not from a specific configuration.
e.g. a key-based verification scenario's purpose is like: an artifact was signed by cosign using a local key is sent to cosign verifier for verification. When a signature comes in, cosign-verifier does not know if it's a keyless or keyed signature from its name. After you confirm the scenario, then we will know what input will be provided to the cosign-verifier.

Reply:
As discussed, ratify verifier interface implementation details would be elaborate in another document in another PR and this document would focus on the background knowledge clarification as how cosign/sigstore-go library is designed and its input/output.